03-events-forms-refs-life-cycle/events/custom_link/src/App.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+import './App.css';
+import CustomLink from './CustomLink'
+
+class App extends Component {
+   
+  constructor(props){
+    super(props);
+
+    this.state ={
+      url1: "https://www.pingmylinks.com/",
+      url2: "https://www.facebook.com/pingmylinks/",
+      url3: "https://www.excitesubmit.com/html/contact.html",
```

Comment:
I don't think this is the best way to model the three URLs, based on how you are handling them in the rest of the file. I would suggest refactoring this to be an array, and having the button text be handled in the `render` method. 

Instead, your state could look like this:

```js
this.state = {
   urls: ["https://www.pingmylinks.com/", "https://www.facebook.com/pingmylinks/", "https://www.excitesubmit.com/html/contact.html"],
   disabled: false
}
```